spark/dl/src/main/scala/com/intel/analytics/bigdl/optim/DistriOptimizer.scala

```diff
@@ -198,12 +199,18 @@ object DistriOptimizer {
               b += 1
             }
           })
-          Engine.default.sync(tasks)
+          if (cached.zeroGradStarted) {
+            assert(cached.tasks.length == _subModelNumber + 1)
```

Comment:
I think it would be better if we give explanation about what we expected and what should we do if assert failed.